### PR TITLE
Added vlan in range check for flow statistics

### LINKED ISSUE #5067: Add VLAN validation for VLAN stat gathering feature
Validation for VLAN value should be added for the following APIs:
- POST https://{host}/v2/flows/{flow_id}/
- PATCH https://{host}/v2/flows/{flow_id}/ 
- PUT https://{host}/v2/flows/{flow_id}/

The allowed VLAN value should be in the range 1 - 4094. Otherwise, an error should be returned

### PR DESCRIPTION
**Pull request by DejfCold**
Closes #5067

### CODE REVIEW

**Review by niksv — COMMENTED** [review]
In general looks good. Please 
1. fix minor comments
2. In PR description please add `Closes #5071`. This line will help github to close the bug when PR will be merged

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidator.java [hunk 1]
```diff
@@ -65,6 +65,12 @@
  */
 public class FlowValidator {
 
+    @VisibleForTesting
+    static final int VLAN_LOWER_BOUND = 1;
+
+    @VisibleForTesting
+    static final int VLAN_UPPER_BOUND = 4094;
```
[the remark this answers is not in the document]
Comment (DejfCold) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidator.java]: Ok, renamed

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidator.java [hunk 2]
```diff
@@ -181,6 +188,23 @@ void checkFlowForCorrectOuterVlansWithVlanStatistics(RequestedFlow flow) throws
...
+
+        if (vlanStatistics == null || vlanStatistics.isEmpty()) {
+            return;
+        }
+
+        boolean isAnyVlanOutsideOfBounds = vlanStatistics.stream()
+                .anyMatch(it -> it < VLAN_LOWER_BOUND || it > VLAN_UPPER_BOUND);
+
+        if (isAnyVlanOutsideOfBounds) {
+            throw new InvalidFlowException("To collect vlan statistics, the vlan IDs must be in valid range",
```
[the remark this answers is not in the document]
Comment (DejfCold) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidator.java]: Done.

File: src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidatorTest.java [hunk 3]
```diff
@@ -199,6 +201,18 @@ public void shouldFailOnSpecifiedBothOuterVlansAndVlanStatistics() throws Invali
...
+        RequestedFlow flow = buildFlow(VLAN_1, VLAN_2, newHashSet(STATS_VLAN_TOO_LOW));
+        flowValidator.checkFlowForVlanStatisticsInCorrectRange(flow);
+    }
+
+    @Test(expected = InvalidFlowException.class)
+    public void shouldFailOnTooHighVlanStatistics() throws InvalidFlowException {
+        RequestedFlow flow = buildFlow(VLAN_1, VLAN_2, newHashSet(STATS_VLAN_TOO_HIGH));
+        flowValidator.checkFlowForVlanStatisticsInCorrectRange(flow);
+    }
+
```
Comment (niksv) [inline src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidatorTest.java:215]: Also I think i worth to add tests for case `if (vlanStatistics == null || vlanStatistics.isEmpty())`

File: src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidatorTest.java [hunk 4]
```diff
@@ -199,6 +201,18 @@ public void shouldFailOnSpecifiedBothOuterVlansAndVlanStatistics() throws Invali
         flowValidator.checkFlowForCorrectOuterVlansWithVlanStatistics(flow);
     }
 
+    @Test(expected = InvalidFlowException.class)
+    public void shouldFailOnTooLowVlanStatistics() throws InvalidFlowException {
```
Comment (niksv) [inline src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidatorTest.java:205]: It's funny but just today (on daily sync) we decided to do not use "should" in test names. 
Common pattern must be `findBlaBlaBlaTest` and `failOnTooLowVlanStatisticsTest`

File: src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidatorTest.java (hunk 3 above)
Comment (DejfCold) [reply src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidatorTest.java:215]: I'd say those are already there
https://github.com/telstra/open-kilda/blob/0f1b8f9c93bcf9ca60d5197f5b96254a8179c797/src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidatorTest.java#L184-L194

Comment (DejfCold) [reply src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidatorTest.java:215]: added for both

File: src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidatorTest.java (hunk 4 above)
Comment (DejfCold) [reply src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidatorTest.java]: Ok, renamed.

**Review by niksv — APPROVED** [review]
